runtime/expr/function/grep.go

```diff
@@ -0,0 +1,43 @@
+package function
+
+import (
+	"fmt"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/runtime/expr"
+	"golang.org/x/text/unicode/norm"
+)
+
+type Grep struct {
+	grep   expr.Evaluator
+	grestr string
+	zctx   *zed.Context
+}
+
+func (g *Grep) Call(ectx zed.Allocator, vals []zed.Value) *zed.Value {
+	patternArg, inputArg := vals[0], vals[1]
```

Comment:
Nits: We usually give these things a "Val" suffix. And no need to copy them.
```suggestion
	patternVal, inputVal := &vals[0], &vals[1]
```